manager_janitor.go

```diff
@@ -519,28 +643,65 @@ func classifyAddRemoveRestartPIndexes(mgr *Manager, addPlanPIndexes []*PlanPInde
 					planPIndexesToAdd = append(planPIndexesToAdd, planPIndexes...)
 					continue
 				}
-				if pindexImplType.AnalyzeIndexDefUpdates != nil &&
+				if phaseChange(configAnalyzeReq) == HIBERNATE_PINDEX {
+					pidxList := getPIndexesToRestart(pindexes, planPIndexes)
+					for _, pidx := range pidxList {
+						pindexesToTransition = append(pindexesToTransition,
+							&pindexStatusChangeReq{
+								pindex:         pidx.pindex,
+								planPIndexName: pidx.planPIndexName,
+								reqType:        HIBERNATE_PINDEX,
+							})
+					}
+					continue
+				}
+				if phaseChange(configAnalyzeReq) == PHASE_CHANGE {
```

Comment:
these two blocks could be collapsed into one by initializing 
reqType: reqTypeValueFromSingleInvocationOfPhaseChangeOutsideTheIfBlocks?

something like 
pc := phaseChange(configAnalyzeReq)

if pc == PHASE_CHANGE || pc == HIBERNATE_PINDEX {
pindexesToTransition = append(pindexesToTransition,
							&pindexStatusChangeReq{
reqType: pc
}

??
}

and apply similar reductions everywhere possbile...